docs/contributing/mentorship.md

```diff
@@ -82,4 +82,43 @@ The example retrospective process looks as follows:
+7. Send an email to the selected mentee asking for confirmation, and if they are happy with this.
+8. Once they reply, fill out the selection form in the relevant portal! :)
+
+#### Recommendations
+
+During mentee selection for our various mentorship programs, we want to follow some best practices to ensure that the mentorship is a fruitful experience for both mentees and mentors. A large part of ensuring this is selecting the most suitable candidate for the mentorship!
+
+This process is hard and requires time, as it involves holistically looking at mentorship applications and trying to ascertain if they would be able to benefit from this mentorship and ideally become a contributor, maintainer or hire in near future.
+
+With several past mentorships, we have been able to distil some learnings from the process, which are listed here as recommendations to mentors:
+
+* Encourage mentees interested in applying by asking them to look through our issues and try to work on one in which they would learn a bit about the codebase! There is no better proof of their motivation to work with us than seeing (meaningful) activity in the open-source project, before selection period.
+* Allocate some time upfront for this process, as you might have a lot of mentee applications to look through.
+  * If possible, book some time on your calendars and discuss the selection process with co-mentors in a particular quarter beforehand.
+* Try to have some prerequisites in mind while shortlisting an application of a potential mentee (CV/questions/proposal/interview),
+  * It is preferable to have mentees from historically underrepresented communities in tech, to ensure diverse contributors in Thanos.
+  * The mentee should be able to allocate enough time to the mentorship.
+    * Thus, having prior important commitments like a full-time job or large educational course load can lead to mentees not having enough time for the mentorship, even if they claim to commit to working > 30hrs a week. This could manifest as mentees not having enough energy to think about a project, skipping meetings, and not being able to utilize the time mentors can spend with them. It is hard to help them in such case. So it is better to practically judge the amount of work a mentee would be able to balance before selecting them.
+    * Generally, we expect our mentees to be university/college students, who would be able to balance both the mentorship and a moderate amount of course load. Commitments larger than this would need to be treated with some caution. Generally, it's only ok to have a full-time job only for more senior engineers, who understand how to organize work, be transparent about mentorship with their employer etc.
+      * In case the mentee is employed full-time, asking for some form of proof (letter/forwarded email from manager) that their employer approves of them working on this mentorship is preferable.
```

Comment:
```suggestion
      * In case the mentee is employed full-time, we recommend asking for some form of proof (letter/forwarded email from manager) that their employer approves of them working on this mentorship is preferable.
```

Reply:
"is preferable" is at the end, so that makes this optional anyway 🙂